spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/InferShape.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn.abstractnn
+
+import com.intel.analytics.bigdl.nn.Graph._
+import com.intel.analytics.bigdl.nn.keras.KerasLayer
+import com.intel.analytics.bigdl.utils.{Node, Shape}
+
+import scala.collection.mutable.ArrayBuffer
+import scala.reflect.ClassTag
+
+trait InferShape {
+
+  private[bigdl] var _inputShapeValue: Shape = null
+
+  private[bigdl] var _outputShapeValue: ArrayBuffer[Shape] = ArrayBuffer[Shape]()
+
+  private[bigdl] def inputShapeValue: Shape = _inputShapeValue
+
+  private[bigdl] def outputShapeValue: ArrayBuffer[Shape] = _outputShapeValue
+
+  // scalastyle:off
+  private[bigdl] def inputShapeValue_=(value: Shape): Unit = {
+    _inputShapeValue = value
+  }
+
+  private[bigdl] def outputShapeValue_=(value: ArrayBuffer[Shape]): Unit = {
+    _outputShapeValue = value
+  }
+  // scalastyle:on
+
+  private[bigdl] def compatibleWithKeras(): Boolean = true
```

Comment:
The meaning of the method name is a little unclear. What about compatibleWithKerasAPI? or kerasAPI?

Reply:
not the same meaning of "kerasapi" i.e torch layer may also compatible with Keras
Maybe it would be more clear if I rename it to "isCompatibleWithKeras"